src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -51,46 +51,51 @@
  * @author Peter Güttinger
  */
 @Name("Tool")
-@Description("The item a player is holding.")
-@Examples({"player is holding a pickaxe",
-		"# is the same as",
-		"player's tool is a pickaxe",
-		"player's off hand tool is shield #Only for Minecraft 1.9"})
+@Description("The item an entity is holding in their main or off hand.")
+@Examples({"player's tool is a pickaxe",
+	"player's off hand tool is a shield",
+	"set tool of all player's to diamond sword",
+	"set offhand tool of target entity to a bow"})
 @Since("1.0")
 public class ExprTool extends PropertyExpression<LivingEntity, Slot> {
 	static {
-		Skript.registerExpression(ExprTool.class, Slot.class, ExpressionType.PROPERTY, "[the] (tool|held item|weapon) [of %livingentities%]", "%livingentities%'[s] (tool|held item|weapon)");
+		Skript.registerExpression(ExprTool.class, Slot.class, ExpressionType.PROPERTY,
+			"[the] (0¦(tool|held item|weapon)|1¦(off[ ]hand (tool|item)|shield[ item])) [of %livingentities%]",
+			"%livingentities%'[s] (0¦(tool|held item|weapon)|1¦(off[ ]hand (tool|item)|shield[ item]))");
 	}
-	
+
+	private boolean offHand;
+
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
```

Comment:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
```
You may remove final from all method parameters and effectively final variables, as it is verbose and not needed